src/commands/list-issues.test.ts

```diff
@@ -0,0 +1,78 @@
+import listIssues from './list-issues';
+import { IMigrationContext } from '../migration-context';
+import mockAdapter from '../adapters/adapter.mock';
+import mockLogger from '../logger/logger.mock';
+import { getIssueTrackerFile } from '../util/persisted-data';
+
+jest.mock('fs-extra', () => {
+    return {
+        // Mock other methods as needed
+        readFile: jest.fn().mockResolvedValue('[{"issue Number": "3", "issue Title": "new title"}]'),
+    };
+});
+
+jest.mock('../util/persisted-data');
+
+describe('list-issue command', () => {
+    let mockContext: IMigrationContext;
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+
+        mockContext = {
+            shepherd: {
+                workingDirectory: 'workingDirectory',
+            },
+            migration: {
+                migrationDirectory: 'migrationDirectory',
+                spec: {
+                    id: 'id',
+                    title: 'title',
+                    adapter: {
+                        type: 'adapter',
+                    },
+                    hooks: {},
+                    issues: {
+                        title: 'issue title',
+                        description: 'issue description',
+                        state: 'open',
+                        state_reason: 'not_planned',
+                        labels: ['bug']
+                    }
+                },
+                workingDirectory: 'workingDirectory',
+                selectedRepos: [{ name: 'selectedRepos' }],
+                repos: [{ name: 'selectedRepos' }],
+                upstreamOwner: 'upstreamOwner',
+            },
+            adapter: mockAdapter,
+            logger: mockLogger,
+        };
+    });
+
+    it('should list issues when the command is invoked',
+        async () => {
+
+            (getIssueTrackerFile as jest.Mock).mockResolvedValueOnce(
+                [{
+                    issueNumber: '7',
+                    title: 'this is my first updated issue',
+                    owner: 'upstreamOwner',
+                    repo: 'selectedRepos'
+                },
+                {
+                    issueNumber: '8',
+                    title: 'this is my first updated issue',
+                    owner: 'newOwner1',
+                    repo: 'newRepo1'
+                },
+            
+            ]
+            );
+
+            await listIssues(mockContext);
+
+            expect(process.stdout.write).toEqual(expect.any(Function))
```

Comment:
process.stdout.write is generally a function, unless it's overwritten.  I imagine a snapshot test here would be effective as we would simply need to assert that the table rendered via process.stdout.write matches the snapshot.

Reply:
the snapshot for process.stdout doesn't seems to be working . I mocked the process.stdout function and validating if its being called In the command execution

Reply:
Looks like this was resolved, however, the content of the table has values of undefined, apart from the column headings.  Should we expect to have non-null values here?

Reply:
yes passed the data and the table shows value now